### PR TITLE
Updated main terraform script to include transcriptome workflow

### PR DESCRIPTION
**Pull request by skanwal**
- Added WTS workflow ID (for dev and prod)
- Added WTS workflow version (for dev and prod)
- Declared inputs (for dev and prod)
    - Mirrored data in `umccr-refdata-prod` from `umccr-refdata-dev`
    
**TO DO**
The `resource` section will need an update. Can I leave that with you?
Happy to discuss if there are any required changes.

### CODE REVIEW

File: terraform/stacks/umccr_data_portal/workflow/main.tf
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:205]: Possible to harmonize using underscore as space separator? Like `sample_name` instead.

File: terraform/stacks/umccr_data_portal/workflow/main.tf
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:237]: Ditto `sample_name`

**Review by victorskl — COMMENTED** [review]
> The resource section will need an update. Can I leave that with you?

Sure, no problem. I will follow up tomorrow.

File: terraform/stacks/umccr_data_portal/workflow/main.tf
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:237]: I don't think sample name is an input into this workflow

```
$ cwltool --make-template workflows/dragen-transcriptome-pipeline/3.7.5/dragen-transcriptome-pipeline__3.7.5.cwl 2>/dev/null | yq eval --tojson
{
  "reference_tar": {
    "class": "File",
    "path": "a/file/path"
  },
  "reference_fasta": {
    "class": "File",
    "path": "a/file/path"
  },
  "protein_domains": {
    "class": "File",
    "path": "a/file/path"
  },
  "output_file_prefix": "a_string",
  "output_directory": "a_string",
  "fastq_list_rows": [
    {
      "rgsm": "a_string",
      "rglb": "a_string",
      "rgid": "a_string",
      "read_2": {
        "class": "File",
        "path": "a/file/path"
      },
      "read_1": {
        "class": "File",
        "path": "a/file/path"
      },
      "lane": 0
    }
  ],
  "enable_rna_quantification": false,
  "enable_rna_gene_fusion": false,
  "enable_map_align_output": false,
  "enable_duplicate_marking": false,
  "cytobands": {
    "class": "File",
    "path": "a/file/path"
  },
  "contigs": "a_string",
  "blacklist": {
    "class": "File",
    "path": "a/file/path"
  },
  "annotation_file": {
    "class": "File",
    "path": "a/file/path"
  }
}
```

Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf:237]: But just on this, we do now have a standard now for inputs and outputs that just allows lowercase, underscores and numbers for both workflows and tools. We've implemented this in the last iteration of cwl-ica and so are just teething out / updating any tools / workflows that violate this rule.

**Comment by alexiswl** [comment]
As of an hour ago I have added 'production_workflows' as a project in the cwl-ica repo and then git versioned controlled this and our bcl-convert and dragen-somatic pipelines.  We can now replace the acl hack that spans a workflow over both dev and prod. Will create a pull request and merge that branch with this branch to resolve merge conflicts

File: terraform/stacks/umccr_data_portal/workflow/main.tf
Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/main.tf]: Yes - we have switched to using `output_file_prefix`. I have removed `sample-name`.

**Comment by skanwal** [comment]
We need to also discuss how we arrange output of this workflow.
Currently Dragen step outputs are going under `output_directory` input.
The other outputs are related to fusion step - that are for now going under `outputDirectory`, defined in engine parameters.

**Comment by alexiswl** [comment]
> We need to also discuss how we arrange output of this workflow.
> Currently Dragen step outputs are going under `output_directory` input.
> The other outputs are related to fusion step - that are for now going under `outputDirectory`, defined in engine parameters.

Dragen outputs will also go under the `engineParameters.outputDirectory`. They will be placed under `engineParameters.outputDirectory/inputs.output_directory`

If you would prefer that the fusion files have their own separate folder you may need to have these files be written to a folder (like dragen) and also have the output folder of these fusions files as an output of the workflow. You may keep the file outputs as outputs too.

File: terraform/stacks/umccr_data_portal/workflow/main.tf
Comment (victorskl) [reply terraform/stacks/umccr_data_portal/workflow/main.tf]: Thanks guys!

**Review by victorskl — APPROVED** [review]
LGTM from my side!

Action items:
- @alexiswl please approve if this is looking good
- @skanwal please merge if this is good and, Alexis is ok

**Comment by skanwal** [comment]
Sounds good. I'll wait for the approval and then merge.

**Comment by alexiswl** [comment]
@skanwal did you want to try merge your files from the arriba steps in the wts pipeline first?

Would be something like so:

```yaml
steps:
  
   run_dragen_transcriptome_step:
     ...
   arriba_fustion_step:
     ...
   arriba_drawing_step:
     ...
    # Create arriba output directory
   create_arriba_output_directory:
      label: create arriba output directory
      doc: |
        Create an output directory to contain the arriba files
      in:
        input_files:
          source:
            - arriba_fusion_step/fusions
            - arriba_fusion_step/discarded_fusions
            - arriba_drawing_step/output_pdf
        output_directory_name:
          valueFrom: "arriba_outputs"
      out:
        - output_directory
      run: ../../../tools/custom-create-directory/1.0.0/custom-create-directory__1.0.0.cwl
```

Outputs would then be as follows:

```yaml
outputs:
  dragen_transcriptome_output_directory:
    label: dragen transcriptome output directory
    doc: |
      The output directory containing all transcriptome output files
    type: Directory
    outputSource: run_dragen_transcriptome_step/dragen_transcriptome_directory
  arriba_output_directory:
    label: arriba output directory
    doc: |
      Output directory for arriba files
    type: Directory
    outputSource: create_arriba_output_directory/output_directory
```

**Comment by skanwal** [comment]
@alexiswl - updated and tested workflow with merged outputs from arriba.

**Review by alexiswl — APPROVED** [review]

**Pull request merged**
